### PR TITLE
[rv_core_ibex, testutils] Refactor to return status_t

### PR DESCRIPTION
**Pull request by engdoreis**
See https://github.com/lowRISC/opentitan/issues/17495

Fix https://github.com/lowRISC/opentitan/issues/17545

### CODE REVIEW

File: sw/device/lib/testing/rv_core_ibex_testutils.c [hunk 1]
```diff
@@ -12,18 +12,18 @@
 bool rv_core_ibex_testutils_is_rnd_data_valid(
     const dif_rv_core_ibex_t *rv_core_ibex) {
   dif_rv_core_ibex_rnd_status_t rnd_status;
-  CHECK_DIF_OK(dif_rv_core_ibex_get_rnd_status(rv_core_ibex, &rnd_status));
-  return rnd_status & kDifRvCoreIbexRndStatusValid;
+  dif_result_t res = dif_rv_core_ibex_get_rnd_status(rv_core_ibex, &rnd_status);
+  return res == kDifOk && rnd_status & kDifRvCoreIbexRndStatusValid;
```
Comment (pamaury) [inline sw/device/lib/testing/rv_core_ibex_testutils.c:16]: I suggest parenthesis here and maybe can explicit test:
 `res == kDifOk && (rnd_status & kDifRvCoreIbexRndStatusValid) != 0`

**Review by timothytrippel — APPROVED** [review]

**Review by pamaury — APPROVED** [review]
Looks good to me modulo the small suggestion in `rv_core_ibex_testutils_is_rnd_data_valid`

File: sw/device/lib/testing/rv_core_ibex_testutils.c (hunk 1 above)
Comment (timothytrippel) [reply sw/device/lib/testing/rv_core_ibex_testutils.c:16]: first comment is valid tho

Comment (engdoreis) [reply sw/device/lib/testing/rv_core_ibex_testutils.c]: Done

**Pull request merged**